tests/testthat/test-pmcmc-nested-parameters.R

```diff
@@ -0,0 +1,430 @@
+  expect_equal(p$summary(),
+    list(p1 = data.frame(name = "a", min = -Inf, max = Inf, discrete = FALSE,
+      type = "varied")))
+})
+
+test_that("pmcmc_nested_parameters prior", {
+  parameters <- list(
+    a = pmcmc_varied_parameter("a", c("p1", "p2"), 1:2,
+        prior = list(function(x) 1, function(x) 2)),
+    b = pmcmc_varied_parameter("b", c("p1", "p2"), 3:4,
+        prior = list(function(x) 3, function(x) 4)),
+    c = pmcmc_parameter("c", 5, prior = function(x) 5),
+    d = pmcmc_parameter("d", 6, prior = function(x) 6))
+  proposal_fixed <- diag(2)
+  proposal_varied <- diag(2) + 1
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied, proposal_fixed)
+
+  init <- p$initial()
+  expect_equal(p$prior(init), set_names(c(15, 17), c("p1", "p2")))
+
+  parameters <- list(
+    a = pmcmc_varied_parameter("a", c("p1", "p2"), 1:2,
+        prior = list(dnorm, dexp)),
+    b = pmcmc_varied_parameter("b", c("p1", "p2"), 3:4,
+        prior = dlnorm),
+    c = pmcmc_parameter("c", 5, prior = function(x) 5),
+    d = pmcmc_parameter("d", 6, prior = function(x) 6))
+
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied, proposal_fixed)
+
+  init <- p$initial()
+  expect_equal(p$prior(init),
+    set_names(c(11 + dnorm(1) + dlnorm(3), 11 + dexp(2) + dlnorm(4)),
+        c("p1", "p2")))
+
+  # 1 varied 1 pop
+  parameters <- list(a = pmcmc_varied_parameter("a", "p1", 1, prior = dnorm))
+  proposal_varied <- diag(1)
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied)
+  init <- p$initial()
+  expect_equal(p$prior(init), set_names(dnorm(1), "p1"))
+
+  # 1 varied 1 fix
+  parameters <- list(
+    a = pmcmc_varied_parameter("a", c("p1", "p2"), 1, prior = dnorm),
+    b = pmcmc_parameter("b", 1, prior = dexp))
+  proposal_varied <- diag(1)
+  proposal_fixed <- diag(1)
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied, proposal_fixed)
+  init <- p$initial()
+  prior <- dnorm(1) + dexp(1)
+  expect_equal(p$prior(init), set_names(rep(prior, 2), c("p1", "p2")))
+})
+
+test_that("pmcmc_nested_parameters model/transform", {
+  parameters <- list(
+    a = pmcmc_varied_parameter("a", c("p1", "p2"), 1:2,
+        prior = list(function(x) 1, function(x) 2)),
+    b = pmcmc_varied_parameter("b", c("p1", "p2"), 3:4,
+        prior = list(function(x) 3, function(x) 4)),
+    c = pmcmc_parameter("c", 5, prior = function(x) 5),
+    d = pmcmc_parameter("d", 6, prior = function(x) 6))
+  proposal_fixed <- diag(2)
+  proposal_varied <- diag(2) + 1
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied, proposal_fixed)
+
+  expect_equal(p$model(p$initial()), apply(p$initial(), 1, as.list))
+
+  p <- pmcmc_nested_parameters$new(parameters, proposal_varied, proposal_fixed,
+    transform = function(x) as.list(log(x)))
+
+  expect_equal(p$model(p$initial()),
+                apply(p$initial(), 1, function(x) as.list(log(x))))
+})
+
+test_that("pmcmc_nested_parameters propose", {
```

Comment:
For new code it's always hard, but as things progress it's nice to break down test blocks until you can write a short sentence about what the test is testing. So something like:

```
test_that("pmcmc_nested_parameters propose changes subset of theta" {
```

and then include the test. If the test blocks then start getting repetitive, farm the setup code into a helper- file